src/app/api/comment/comment.controller.ts

```diff
@@ -0,0 +1,39 @@
+import { NextRequest, NextResponse } from 'next/server'
+import authenticate from '../core/utils/authenticate'
+import { CommentService } from './comment.service'
+import { CreateCommentSchema, UpdateCommentSchema } from '@/types/dto/comment.dto'
+import httpStatus from 'http-status'
+import { IdParams } from '../core/types/api'
+
+export const createComment = async (req: NextRequest) => {
+  const user = await authenticate(req)
+
+  const commentService = new CommentService(user)
+
+  const data = CreateCommentSchema.parse(await req.json())
+
+  const comment = await commentService.createComment(data)
+
+  return NextResponse.json({ comment }, { status: httpStatus.CREATED })
+}
+
+export const deleteComment = async (req: NextRequest, { params: { id } }: IdParams) => {
+  const user = await authenticate(req)
+
+  const commentService = new CommentService(user)
+
+  await commentService.deleteComment(id)
+
+  return NextResponse.json({ message: 'Comment deleted!' }, { status: httpStatus.OK })
```

Comment:
the response code should be 204 for delete. and please use HttpStatus package instead of hard code